prometheus/collect_consumer_groups.go

```diff
@@ -49,6 +51,71 @@ func (e *Exporter) collectConsumerGroups(ctx context.Context, ch chan<- promethe
 				group.State,
 				strconv.FormatInt(int64(coordinator), 10),
 			)
+
+			// iterate all members and build two maps:
+			// - {topic -> number-of-consumers}
+			// - {topic -> number-of-partitions-assigned}
+			topicConsumers := make(map[string]int)
+			topicPartitionsAssigned := make(map[string]int)
+			membersWithEmptyAssignment := 0
+			var failedAssignmentsDecode int64 = 0
+			for _, member := range group.Members {
+				var kassignment kmsg.GroupMemberAssignment
+				if err := kassignment.ReadFrom(member.MemberAssignment); err != nil {
+					e.logger.Debug("failed to decode consumer group member assignment, internal kafka error",
+						zap.Error(err),
+						zap.String("group_id", group.Group),
+						zap.String("client_id", member.ClientID),
+						zap.String("member_id", member.MemberID),
+						zap.String("client_host", member.ClientHost),
+					)
+					failedAssignmentsDecode++
+				} else {
+					if len(kassignment.Topics) == 0 {
+						membersWithEmptyAssignment++
+					}
+					for _, topic := range kassignment.Topics {
+						topicConsumers[topic.Topic]++
+						topicPartitionsAssigned[topic.Topic] += len(topic.Partitions)
+					}
+				}
+			}
+			if failedAssignmentsDecode > 0 {
+				e.logger.Error("failed to decode consumer group member assignment, internal kafka error",
+					zap.Error(err),
+					zap.String("group_id", group.Group),
+					zap.String("assignment_decode_failures", strconv.FormatInt(failedAssignmentsDecode, 10)),
+				)
+			}
+			// number of members with no assignment in a stable consumer group
+			if membersWithEmptyAssignment > 0 && group.State == "Stable" {
```

Comment:
I'm wondering whether we actually want that condition again (or just report it as zero everytime where this is case - regardless of the current group state). I assume that you have a certain usecase in mind which is why you added these metrics in the first place. Do you care to explain them so that I can understand the idea?

Reply:
Right - superseded by https://github.com/cloudhut/kminion/issues/108